functions/Copy-DbaStartupProcedure.ps1

```diff
@@ -0,0 +1,178 @@
+                    SourceServer      = $sourceServer.Name
+                    DestinationServer = $destServer.Name
+                    Name              = $currentProcName
+                    Schema            = $currentProcSchema
+                    Status            = $null
+                    Notes             = $null
+                    Type              = "Startup Stored Procedure"
+                    DateTime          = [DbaDateTime](Get-Date)
+                }
+
+                if ($Procedure -and ($Procedure -notcontains $currentProcName)) {
+                    continue
+                }
+
+                if ($ExcludeProcedure -and ($ExcludeProcedure -contains $currentProcName)) {
+                    continue
+                }
+
+                if ($destStartupProcs.Name -contains $currentProcName) {
+                    if ($force -eq $false) {
+                        $copyStartupProcStatus.Status = "Skipped"
+                        $copyStartupProcStatus.Notes = "Already exists on destination"
+                        $copyStartupProcStatus | Select-DefaultView -Property DateTime, SourceServer, DestinationServer, Name, Type, Status, Notes -TypeName MigrationObject
+
+                        Write-Message -Level Verbose -Message "Startup procedure $currentProcFullName exists at destination. Use -Force to drop and migrate."
+                        continue
+                    } else {
+                        if ($Pscmdlet.ShouldProcess($destInstance, "Dropping startup procedure $currentProcFullName and recreating")) {
+                            try {
+                                Write-Message -Level Verbose -Message "Dropping startup procedure $currentProcFullName"
+                                $destServer.Query("DROP PROCEDURE [$($currentProcSchema)].[$($currentProcName)]")
+                            } catch {
+                                $copyStartupProcStatus.Status = "Failed"
+                                $copyStartupProcStatus.Notes = $_
+                                $copyStartupProcStatus | Select-DefaultView -Property DateTime, SourceServer, DestinationServer, Name, Type, Status, Notes -TypeName MigrationObject
+
+                                Stop-Function -Message "Issue dropping startup procedure" -Target $currentProcFullName -ErrorRecord $_ -Continue
+                            }
+                        }
+                    }
+                }
+
+                if ($Pscmdlet.ShouldProcess($destInstance, "Creating startup procedure $currentProcFullName")) {
+                    try {
+                        Write-Message -Level Verbose -Message "Copying startup procedure $currentProcFullName"
+                        #$proc = Get-DbaDbStoredProcedure -SqlInstance $sourceServer -Database master | Where-Object {$_.Schema -eq $currentProcSchema -and $_.Name -eq $currentProcName}
+                        $sql = $currentProc.Definition | Out-String
```

Comment:
This will work for now but only has an initial version of the command.

The problem with using the `Definition` metadata is it has a limit on characters, the full definition of complex/long winded procedures will be cut off. [ref](https://sqlblog.org/2011/11/08/t-sql-tuesday-24-dude-wheres-the-rest-of-my-procedure).